From 1afa05de7d93c51a4adcea1633013ad667b65f84 Mon Sep 17 00:00:00 2001
Message-ID: <1afa05de7d93c51a4adcea1633013ad667b65f84.1732384228.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732384228.git.sharlatanus@gmail.com>
References: <cover.1732384228.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 21 Nov 2024 16:06:37 +0000
Subject: [PATCH 24/46] gnu: python-astrodendro: Update to 0.3.1.

* gnu/packages/astronomy.scm (python-astrodendro): Update to 0.3.1.
[source]: Swap to PyPI tarball containing a fresh release.
[phases]: Remove 'set-env-version, no longer required.

Change-Id: I3180990d8a12747b7b301680a705f956a7f47e0c
---
 gnu/packages/astronomy.scm | 26 +++++---------------------
 1 file changed, 5 insertions(+), 21 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 7f020a1ba9..78e9d0c156 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -1808,32 +1808,16 @@ (define-public python-astroalign
     (license license:expat)))
 
 (define-public python-astrodendro
-  ;; XXX: 0.2.0 was released 8 years ago and not compatible with Python 3.10,
-  ;; use the latest commit on master branch, see
-  ;; <https://github.com/dendrograms/astrodendro/issues/189>.
-  (let ((commit "3181c36489caa4c62b36b7ec8e06f4da65af6fda")
-        (revision "0"))
     (package
       (name "python-astrodendro")
-      (version (git-version "0.2.0" revision commit))
+      (version "0.3.1")
       (source
        (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://github.com/dendrograms/astrodendro")
-               (commit commit)))
+         (method url-fetch)
+         (uri (pypi-uri "astrodendro" version))
          (sha256
-          (base32 "1imahzl0g8r1lxyiz9wi9q4zk519yxgn6qs1nb0ybb34sipjs4kz"))
-         (file-name (git-file-name name version))))
+          (base32 "0pav2rq5q0wyr38g6z8ai4z2iqqc9x04iwll158yvkvgnv352m0i"))))
       (build-system pyproject-build-system)
-      (arguments
-       (list
-        #:phases
-        #~(modify-phases %standard-phases
-            (add-before 'build 'set-env-version
-              (lambda _
-                (setenv "SETUPTOOLS_SCM_PRETEND_VERSION"
-                        #$(version-major+minor+point version)))))))
       (native-inputs
        (list python-pytest
              python-setuptools
@@ -1849,7 +1833,7 @@ (define-public python-astrodendro
       (description
        "This package provides an way to compute dendrograms of observed or
 simulated Astronomical data in Python.")
-      (license license:expat))))
+      (license license:expat)))
 
 (define-public python-astroml
   (package
-- 
2.46.0

